git-hub

```diff
@@ -81,21 +137,325 @@ def push():
+    to_write = to_write + 'branch = "{0}" \n'.format(response.get('head').get('ref'))
+    to_write = to_write + 'comment = "{0}" \n'.format(pr.title)
+    to_write = to_write + 'commits = "{0}" \n'.format(response.get('commits'))
+    to_write = to_write + 'modified = "{0}" \n'.format(pr.updated_at)
+    to_write = to_write + 'created_at = "{0}" \n'.format(response.get('created_at'))
+    to_write = to_write + 'updated_at = "{0}" \n'.format(response.get('updated_at'))
+    to_write = to_write + 'closed_at = "{0}" \n'.format(response.get('closed_at'))
+    to_write = to_write + 'merged_at = "{0}" \n'.format(response.get('merged_at'))
+    reviewers = ""
+    for reviewer in list(response.get('requested_reviewers')):
+        reviewers = reviewers + reviewer.get('login')+", "
+    to_write = to_write + 'reviewers = "{0}" \n'.format(reviewers[:-2])
+    assignees = ""
+    for assignee in list(response.get('assignees')):
+        assignees = assignees + assignee.get('login')+", "
+    to_write = to_write + 'assignee = "{0}" \n'.format(assignees[:-2])
+    if(pr.milestone):
+        to_write = to_write + 'milestone = "{0}" \n'.format(pr.milestone.title)
+    else:
+        to_write = to_write + 'milestone = "" \n'
+
+    issue_url = response.get('issue_url')       #fetches labels attached to pull requests
+    labels = ""
+    issue_request = Request(issue_url)
+    issue_response = urlopen(issue_request)
+    issue_response = json.loads(issue_response.read().decode('utf-8'))
+    labels_dict = issue_response.get('labels')
+    for label in list(labels_dict):
+        labels = labels + label.get('name')+", "
+    to_write = to_write + 'labels = "{0}" \n'.format(labels[:-2])
+
+    comment_url = response.get('comments_url')       #fetches latest activity (comment) date
+    comment_count = 0
+    most_recent = ""
+    self_comment = False
+    comment_request = Request(comment_url)
+    comment_response = urlopen(comment_request)
+    comment_dict = json.loads(comment_response.read().decode('utf-8'))
+    for c in list(comment_dict):
+        comment_count = comment_count + 1
+        date = c.get('updated_at')
+        if most_recent == "" or date > most_recent:
+            most_recent = date
+        commenter = c.get('user').get('login')
+        if commenter == username:
+            self_comment = True
+    to_write = to_write + 'comment_count = "{0}" \n'.format(comment_count)
+    to_write = to_write + 'most_recent_comment = "{0}" \n'.format(most_recent)
+    to_write = to_write + 'self_comment = "{0}" \n'.format(self_comment)
+    return to_write
+
+
+def sync():
+    """Updates and saves pull-requests in pull-requests.toml in the .git folder. """
+    username, repo, remotes = login()
+    token = get_token()
+
+    # saves pr into toml file
+    try:
+        g = github.Github(token)
+        path_git = path_to_git()
+        path_github = pjoin(path_git, 'git-hub')
+        path_prs = pjoin(path_github, 'pull-requests.toml')
```

Comment:
These three lines appear repeated all over the place; refactor into a utility function.